source/libtomo/misc/CMakeLists.txt

```diff
@@ -1,7 +1,23 @@
 set(HEADERS "${tomopy_SOURCE_DIR}/include/libtomo/morph.h"
-            "${tomopy_SOURCE_DIR}/include/libtomo/remove_ring.h")
+            "${tomopy_SOURCE_DIR}/include/libtomo/remove_ring.h"
+            "${tomopy_SOURCE_DIR}/include/libtomo/median_filt3d.h")
 
-tomopy_add_library(tomo-misc SHARED morph.c remove_ring.c ${HEADERS})
+tomopy_add_library(tomo-misc SHARED 
+		   morph.c
+		   remove_ring.c
+		   median_filt3d.c
+		   utils.c
+		   utils.h	   
+		   ${HEADERS})
+
+find_package(OpenMP REQUIRED)
+if (OPENMP_FOUND)
+    set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")
+    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
+    set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${OpenMP_EXE_LINKER_FLAGS} ${OpenMP_CXX_FLAGS}")
+    set (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${OpenMP_SHARED_LINKER_FLAGS} ${OpenMP_CXX_FLAGS}")
+    set (CMAKE_STATIC_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${OpenMP_STATIC_LINKER_FLAGS} ${OpenMP_CXX_FLAGS}")
+endif()
```

Comment:
Actually, we might try this:

```cmake
find_package(OpenMP REQURIED)
if(OpenMP_C_FOUND)
    target_link_libraries(MyTarget PUBLIC OpenMP::OpenMP_C)
endif()
```

Reply:
hi, just been trying to benchmark some things here and noticed that your suggestion above doesn't seem to enable multithreading unfortunately (cmake ver.3.25). Reverted to the original version brought it back again. It does work for you, however? What could be the problem? thanks

Reply:
It should work for [any CMake >= 3.9 for Linux](https://cliutils.gitlab.io/modern-cmake/chapters/packages/OpenMP.html). The benchmark I shared is evidence that it works for me because otherwise, your implementation wouldn't speed up with increasing number of cores. Can you share your build logs?

I have also added a new suggestion to the code which removes the conditional, OpenMP_C_FOUND. Maybe CMake isn't detecting OpenMP for C, but for other languages then skipping the link?

Do you have build logs you could share?

Reply:
Also, is there a minimum OpenMP version that these features require?

Reply:
thanks, here are my logs bellow, also OpenMP_C gets detected successfully. Those openmp features were already there since version 3.0.
![Screenshot from 2023-01-10 10-27-52](https://user-images.githubusercontent.com/6651977/211526783-bda11d6d-10d8-45fe-a4d1-03035d7b1332.png)
 but the multithreading is still disabled in my case when I do:
```
find_package(OpenMP REQUIRED COMPONENTS C)
target_link_libraries(tomo-misc PRIVATE OpenMP::OpenMP_C)
```
What gcc do you use? Don't know if that matters, I've got 7.5.

[CMakeOutput.log](https://github.com/tomopy/tomopy/files/10381568/CMakeOutput.log)
[CMakeError.log](https://github.com/tomopy/tomopy/files/10381570/CMakeError.log)

Reply:
I think some project-wide flags were overshadowing the local ones. Try again please.

Reply:
thanks a lot, the recent changed helped.